From 7f8fa91288ad7b7287d403f12910ff50c5be96d1 Mon Sep 17 00:00:00 2001
From: Sushant Butta <quic_sbutta@quicinc.com>
Date: Wed, 31 Jul 2024 18:05:19 +0530
Subject: [PATCH] skb-recycler: Add support to track the original caller

    The chages are ported from the 5.4 kernel from the
    following gerrits:
    https://review-android.quicinc.com/c/oss/kernel/linux-ipq-5.4/+/4915164
    https://review-android.quicinc.com/c/oss/kernel/linux-ipq-5.4/+/5199994
These are required for debug config compilation.

Change-Id: I295212dbca214ff7d5ea749956b2d79d4fd8bc2c
Signed-off-by: Sushant Butta <quic_sbutta@quicinc.com>
---
 include/linux/debugobjects.h | 1 +
 include/linux/mm.h           | 4 ++++
 2 files changed, 5 insertions(+)

diff --git a/include/linux/debugobjects.h b/include/linux/debugobjects.h
index fb9abddbdb7e..02111d2ccfcc 100644
--- a/include/linux/debugobjects.h
+++ b/include/linux/debugobjects.h
@@ -116,6 +116,7 @@ struct debug_obj_trace {
 
 extern void debug_object_trace_init(void *addr, void **stack, size_t size);
 extern void debug_object_trace_free(void *addr);
+extern void debug_object_trace_update(void *addr, void **stack);
 extern void debug_mem_usage_init(void);
 #else
 static inline void debug_mem_usage_init(void) { }
diff --git a/include/linux/mm.h b/include/linux/mm.h
index 0b986dd637f8..16714d06458a 100644
--- a/include/linux/mm.h
+++ b/include/linux/mm.h
@@ -3993,6 +3993,10 @@ unsigned long wp_shared_mapping_range(struct address_space *mapping,
 
 extern int sysctl_nr_trim_pages;
 
+#ifdef CONFIG_DEBUG_MEM_USAGE
+void __wrap_update_call_stack(void *addr);
+#endif
+
 #ifdef CONFIG_PRINTK
 void mem_dump_obj(void *object);
 #else
-- 
2.34.1

